modules/host/update.go

```diff
@@ -74,17 +75,29 @@ func (h *Host) ProcessConsensusChange(cc modules.ConsensusChange) {
 	h.mu.Lock()
 	defer h.mu.Unlock()
 
-	h.blockHeight += types.BlockHeight(len(cc.AppliedBlocks))
-	h.blockHeight -= types.BlockHeight(len(cc.RevertedBlocks))
+	// Adjust the height of the host. The host height is initialized to zero,
+	// but the genesis block is actually height zero. For the genesis block
+	// only, the height will be left at zero.
+	//
+	// Checking the height here eliminates the need to initialize the host to
+	// and underflowed types.BlockHeight.
```

Comment:
I think it's worse to treat the genesis block special. I think it'd be better if genesis block were height 1.

It's nice to have the genesis block get provided by a subscription because it's got diffs, otherwise you'd have to call ProcessGenesisBlock() or something, and that's clunky too. I still prefer the way it's set up here, though you are right it will end up appearing in every module.